### PR TITLE
Fixes issue with filtering with property values of number 0

### PR DESCRIPTION
**Pull request by pushpinder107**
**What**:
https://github.com/kentcdodds/match-sorter/issues/46
The filter does not return data if the value of the key being filtered is the number 0

**Why**:
It is possible that the data may contain properties with values which are numbers (e.g. age) and these properties could have the value 0 as well. Hence, filtering on 0 should be possible.

**How**:
By adding an additional check to make sure the number 0 is not incorrectly rejected by the if statement.

**Checklist**:

- [ ] Documentation "N/A"
- [x] Tests
- [x] Ready to be merged 
- [x] Added myself to contributors table

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -387,7 +387,7 @@ function getItemValues(item, key) {
     value = item[key]
   }
   // concat because `value` can be a string or an array
-  return value ? [].concat(value) : null
+  return value || value === 0 ? [].concat(value) : null
```
Comment (kentcdodds) [inline src/index.js]: I'd rather change this to:

```js
return value == null ? [].concat(value) : null
```

That way we cover null and undefined and all other cases will be concatenate.

**Comment by pushpinder107** [comment]
Nope, validate script fails locally :
Failed at the match-sorter@0.0.0-semantically-released validate script.

**Comment by kentcdodds** [comment]
Hmmm... Looks like your changes are not related to this. It's breaking for me locally without your changes. I'll look into it.

**Comment by kentcdodds** [comment]
I'm not sure what happened, but things were very out of date. I've updated everything and it should work now.

**Comment by kentcdodds** [comment]
Thanks a bunch!

**Pull request merged**